### PR TITLE
SystemMoveIn, SystemPush, and SystemReveal animators

### PR DESCRIPTION
**Pull request by tbaranes**
In this PR, we continue to add more system animators (#126):
- SytemMoveIn
- SystemPush
- SystemReveal

I did one PR for the three because the implementation is exactly the same: one system animator supporting the four directions.

### CODE REVIEW

File: IBAnimatable/SystemMoveInAnimator.swift [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+      self.reverseAnimationType = .SystemMoveIn(direction: .Right)
+      self.interactiveGestureType = .PanFromRight
+    case .Right:
+      self.transitionAnimationType = .SystemMoveIn(direction: .Right)
+      self.reverseAnimationType = .SystemMoveIn(direction: .Left)
+      self.interactiveGestureType = .PanFromLeft
+    case .Top:
+      self.transitionAnimationType = .SystemMoveIn(direction: .Top)
+      self.reverseAnimationType = .SystemMoveIn(direction: .Bottom)
+      self.interactiveGestureType = .PanFromTop
```
Comment (JakeLin) [inline IBAnimatable/SystemMoveInAnimator.swift:34]: I think it should be `. PanFromBottom`, need to fix for line 38, and `SystemPushAnimator` as well.

**Comment by JakeLin** [comment]
👍 except for a smalll comment. Well done, you are so productive.

File: IBAnimatable/SystemMoveInAnimator.swift (hunk 1 above)
Comment (tbaranes) [inline IBAnimatable/SystemMoveInAnimator.swift:34]: Do you think? If that so, we have the wrong implementation of `InteractiveGesture` for each animators. Gonna try it on real device, and get back to you.

Comment (JakeLin) [inline IBAnimatable/SystemMoveInAnimator.swift:34]: The pop gesture should be reverse as push animation. You are right, I implemented `handleGesture` in `PanInteractiveAnimator` wrongly, please merge this PR when you free ready, I fix them all in next PR. Thanks.

**Pull request merged**